containers/map.go

```diff
@@ -22,6 +22,49 @@ func (m *ConcurrentMap[K, V]) Get(key K) (V, bool) {
 	return val, ok
 }
 
+// GetOrCreate returns the existing value for the key if present. Otherwise, it stores and returns the given value.
+// The loaded result is true if the value was loaded, false if stored.
+func (m *ConcurrentMap[K, V]) GetOrCreate(key K, val V) (V, bool) {
+	m.mx.Lock()
```

Comment:
double-checked locking here?

Reply:
Why? We only access map if the lock was held.

Reply:
why not? if there is already an element, we don't even need to acquire the lock, no? or doesn't it work that way in Go for some reason?

Reply:
hmm, interesting, but ok :)